sql/core/src/test/scala/org/apache/spark/sql/execution/SparkSqlParserSuite.scala

```diff
@@ -301,4 +301,10 @@ class SparkSqlParserSuite extends PlanTest {
       "SELECT a || b || c FROM t",
       Project(UnresolvedAlias(concat) :: Nil, UnresolvedRelation(TableIdentifier("t"))))
   }
+
+  test("reset") {
+    assertEqual("reset", ResetCommand(None))
+    assertEqual("reset spark.test.property", ResetCommand(Some("spark.test.property")))
+    assertEqual("reset #$a!", ResetCommand(Some("#$a!")))
```

Comment:
can we check hive's behavior? I think special chars are not allowed in config name and parser should throw exception for this case.